meson.build

```diff
@@ -1,42 +1,47 @@
 # Copyright (c) 2019 Maxim Egorushkin. MIT License. See the full licence in file LICENSE.
 
-# (rm -rf build; meson build; cd build; time ninja -v)
+# (rm -rf build; meson setup build; time ninja -C build -v)
 
 project(
   'atomic_queue', 'cpp',
   license : 'MIT License',
   default_options : ['cpp_std=gnu++14', 'buildtype=release', 'b_ndebug=if-release']
 )
 
-cxx = meson.get_compiler('cpp')
-dl = cxx.find_library('dl', required : true)
-threads = dependency('threads')
-unit_test_framework = dependency('boost', modules : ['unit_test_framework'])
-if get_option('benchmarks')
-  tbb = cxx.find_library('tbb', required : true)
-  xenium = declare_dependency(include_directories : '../xenium')
-  moodycamel = declare_dependency(include_directories : '../')
-endif
+threads_dep = dependency('threads')
 
-atomic_queue = declare_dependency(include_directories : ['include'], dependencies : threads)
+atomic_queue_dep = declare_dependency(include_directories : ['include'], dependencies : threads_dep)
 
-tests_exe = executable(
-  'tests',
-  'src/tests.cc',
-  dependencies : [atomic_queue, unit_test_framework]
-)
-test('tests', tests_exe)
+if get_option('tests')
+  unit_test_framework_dep = dependency('boost', modules : ['unit_test_framework'])
 
-example_exe = executable(
-  'example',
-  'src/example.cc',
-  dependencies : [atomic_queue]
-)
+  tests_exe = executable(
+    'tests',
+    'src/tests.cc',
+    dependencies : [atomic_queue_dep, unit_test_framework_dep],
+  )
+  test('tests', tests_exe)
+
+  example_exe = executable(
+    'example',
+    'src/example.cc',
+    dependencies : [atomic_queue_dep],
+  )
+  test('example', example_exe)
+endif
 
 if get_option('benchmarks')
-  benchmarks_exe = executable(
+  cxx = meson.get_compiler('cpp')
+  dl_dep = cxx.find_library('dl')
+  xenium_dep = declare_dependency(include_directories : '../xenium')
```

Comment:
Building benchmarks with xenium requires `-std=gnu++17` compiler option these days. 

Makefile was updated to build benchmarks only in gnu++17 mode. While keeping gnu++14 mode for unit-tests and the example. But not meson.build.

Reply:
meson.build didn't get enough usage and love, and fell out of date. 

The compiler and linker flags this library requires is `-std=gnu++14 -pthread`.

Reply:
`threads_dep` is probably what adds `-pthread`, I stand corrected.

Reply:
I wonder if a meson fan could help restore `meson.build` into its original fully-functional glory? @stephanlachnit

Reply:
> Makefile was updated to build benchmarks only in gnu++17 mode. While keeping gnu++14 mode for unit-tests and the example. But not meson.build.

Ah, I haven't actually tried compiling the benchmarks, but we can add this check

> `threads_dep` is probably what adds `-pthread`, I stand corrected.

Yes, that is correct

> I wonder if a meson fan could help restore meson.build into its original fully-functional glory?

I can take a look at it see if I can add it to the CI